C-Plus-Plus/sliding_window/1st_negative_no_in_every_window_of_size_k.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+Problem Statement:
+An array and a positive integer k is given, we need to find 1st negative integer for each and every window(contiguous subarray) of size k
+
+
+Brute force:-
+	take two pointers say i,j 
+	Initialise i from 0 and  j from i to k and find the 1st negative.
+	In this , we are doing repetitive work 
+	therefore it's time complexity O(n^2)
+*/
+
+#include<bits/stdc++.h>
+using namespace std;
+int main(){
+	int n;
+	cin>>n;
+	int arr[n];
+	for(int i=0;i<n;i++)
+		cin>>arr[i];
+	int k;
+	cin>>k;
+	//dynamic array for storing negative elements
+	vector<int>list;
+	//dymanice array for storing 1st negative of each window
+	vector<int>ans;
+	int i=0,j=0;
+	//if j non negative then we don.t need it
+	//if negative, store it in array
+	while(j<n){
+		//element is negative then push it into list
+		if(arr[j]<0)
+			list.push_back(arr[j]);
+		//window size not reached then increment j
+		if(j-i+1<k)
+			j++;
+		//on hitting window size
+		else if(j-i+1==k)
+		{	//no negative element in current window 
+			if(list.size()==0)
+			ans.push_back(0);
+			else 
+				{//push the 1st negative of every window
+					ans.push_back(list.front());
+					//negative element is included in list only if it is in starting
+					//we need to remove that element which is nit in window	
+					if(arr[i]==list.front())
+					//delete that	
+					list.erase(list.begin());
+			    }
+
+			    //slide the window
+				i++;
+				j++;
+
+		}
+
+	}
+	for(int i=0;i<ans.size();i++)
+		cout<<ans[i]<<" ";
```

Comment:
Display a message (before this) what you are printing for better user-interface.